From bf0efffb7dbe44abd84bc44c1c5e8509da0a654c Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Sat, 30 Dec 2017 23:56:33 +0100
Subject: [PATCH] msm8974-common: Use QTI HIDL power HAL

 * Let's use the new shiny binderized HAL in hardware/qcom/power

Change-Id: I2abf4ca9b0b5a0cb031f12c400a1159962f5e94c
---
 BoardConfigCommon.mk | 5 +++++
 manifest.xml         | 2 +-
 msm8974.mk           | 3 +--
 3 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/BoardConfigCommon.mk b/BoardConfigCommon.mk
index 80453f5..75a53ed 100644
--- a/BoardConfigCommon.mk
+++ b/BoardConfigCommon.mk
@@ -66,6 +66,11 @@ DEVICE_MATRIX_FILE := device/samsung/msm8974-common/compatibility_matrix.xml
 TARGET_INIT_VENDOR_LIB := libinit_msm8974
 TARGET_RECOVERY_DEVICE_MODULES := libinit_msm8974
 
+# Power
+TARGET_HAS_LEGACY_POWER_STATS := true
+TARGET_HAS_NO_WIFI_STATS := true
+TARGET_USES_INTERACTION_BOOST := true
+
 # Time services
 BOARD_USES_QC_TIME_SERVICES := true
 
diff --git a/manifest.xml b/manifest.xml
index a5f3b63..1fdd417 100644
--- a/manifest.xml
+++ b/manifest.xml
@@ -116,7 +116,7 @@
     </hal>
     <hal format="hidl">
         <name>android.hardware.power</name>
-        <transport arch="32">passthrough</transport>
+        <transport>hwbinder</transport>
         <version>1.0</version>
         <interface>
             <name>IPower</name>
diff --git a/msm8974.mk b/msm8974.mk
index df1d8d1..69c2263 100644
--- a/msm8974.mk
+++ b/msm8974.mk
@@ -135,8 +135,7 @@ PRODUCT_PACKAGES += \
 
 # Power HAL
 PRODUCT_PACKAGES += \
-    android.hardware.power@1.0-impl \
-    power.msm8974
+    android.hardware.power@1.0-service-qti
 
 # RenderScript HAL
 PRODUCT_PACKAGES += \
-- 
2.7.4

